ipfs.h

```diff
@@ -36,24 +37,44 @@ int execute_command_and_save_result(char* file_name, unsigned char* hash_value);
 // @param ctx information to be included in encryption.
 // @param my_file_path path of the file to encrypt.
 // @param hash_value value to send to file system manager.
-int file_encrypt_upload(SST_session_ctx_t* session_ctx, SST_ctx_t* ctx, char* my_file_path, unsigned char* hash_value);
+int file_encrypt_upload(session_key_t* session_ctx, SST_ctx_t* ctx, char* my_file_path, unsigned char* hash_value);
 
 // Download the file in IPFS environment and decrypt the file with sessionkey.
 // @param session_ctx session key to decrypt the file.
 // @param file_name file name to save in my repository.
-void file_download_decrypt(SST_session_ctx_t* session_ctx, char* file_name);
+void file_download_decrypt(session_key_t session_ctx, char* file_name);
 
 // Request the data to file system manager.
 // @param session_ctx session key information to send to file system manager.
 // @param ctx owner information to send to file system manager.
 // @param hash_value value to send to file system manager.
 // @param hash_value_len length of value to send to file system manager.
-void upload_to_file_system_manager(SST_session_ctx_t* session_ctx, SST_ctx_t* ctx, unsigned char* hash_value, int hash_value_len);
+void upload_to_file_system_manager(session_key_t* session_ctx, SST_ctx_t* ctx, unsigned char* hash_value, int hash_value_len);
 
 // Receive the data from file system manager.
 // @param session_ctx session key information to compare with session key received from file system manager.
 // @param ctx information to access the file system manager.
 // @param file_name file name to save the file.
-void download_from_file_system_manager(SST_session_ctx_t* session_ctx, SST_ctx_t* ctx, char* file_name);
+void download_from_file_system_manager(unsigned char* skey_id, SST_ctx_t* ctx, char* file_name);
+
+// Check the session key and request to Auth if you don't have the session key you want.
+// Return the session key information you want.
```

Comment:
```suggestion
// Check if the target session key exists. If the target session key exists, return the session key.
// Otherwise, request and receive the target session key from Auth and return the session key.
```